aws/resource_aws_wafv2_web_acl_test.go

```diff
@@ -367,16 +371,19 @@ func TestAccAwsWafv2WebACL_RuleGroupReferenceStatement(t *testing.T) {
 					resource.TestCheckResourceAttr(resourceName, "name", webACLName),
 					resource.TestCheckResourceAttr(resourceName, "rule.#", "1"),
 					computeWafv2RuleGroupRefStatementIndex(&v, &idx, excludedRules),
-					testCheckResourceAttrWithIndexesAddr(resourceName, "rule.%d.name", &idx, "rule-1"),
-					testCheckResourceAttrWithIndexesAddr(resourceName, "rule.%d.override_action.#", &idx, "1"),
-					testCheckResourceAttrWithIndexesAddr(resourceName, "rule.%d.override_action.0.count.#", &idx, "1"),
-					testCheckResourceAttrWithIndexesAddr(resourceName, "rule.%d.override_action.0.none.#", &idx, "0"),
-					testCheckResourceAttrWithIndexesAddr(resourceName, "rule.%d.statement.#", &idx, "1"),
-					testCheckResourceAttrWithIndexesAddr(resourceName, "rule.%d.statement.0.rule_group_reference_statement.#", &idx, "1"),
+					tfawsresource.TestCheckTypeSetElemNestedAttrs(resourceName, "rule.*", map[string]string{
+						"name":                             "rule-1",
+						"override_action.#":                "1",
+						"override_action.0.count.#":        "1",
+						"rule.%d.override_action.0.none.#": "0",
+						"rule.%d.statement.#":              "1",
+						"rule.%d.statement.0.rule_group_reference_statement.#":              "1",
```

Comment:
you can omit the `rule.%d.` prefix for these 

```suggestion
						"override_action.0.none.#": "0",
						"statement.#":              "1",
						"statement.0.rule_group_reference_statement.#":              "1",
```